src/soc/ibm/power9/istep_18_11.c

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <console/console.h>
+#include <cpu/power/istep_18.h>
+
+// TODO: this one will cange if we add more than one processor
+#define MDMT (1)
+
+static uint32_t calculate_topology_dealy(void)
+{
+  // in simple topology with one proc it is enough to assign 0
+  // with multiple processors this will get more complicated
+  // see calculate_node_link_delay() in Hosboot
+
+  if(MDMT)
+  {
+    uint32_t node_link_delay = 0;
+    uint32_t longest_topology_delay = 0;
+
+    return
+      - longest_topology_delay
+      + node_link_delay
+      + MDMT_TOD_GRID_CYCLE_STAGING_DELAY;
+  }
+
+  write_scom(PU_PB_ELINK_RT_DELAY_CTL_REG, PPC_BITMASK(2, 3));
+  uint64_t l_bus_mode_reg = read_scom(PU_PB_ELINK_DLY_0123_REG);
```

Comment:
Please use `_for_chiplet` versions here.

Reply:
@krystian-hebel This line of code is the equivalent of a [statement bellow](https://github.com/open-power/hostboot/blob/46a489ba7864b1d32c4b37a1223a39f33ad12f66/src/import/chips/p9/procedures/hwp/nest/p9_tod_setup.C#L1332).
```
FAPI_TRY(fapi2::getScom(*(i_tod_node->i_target),
                            l_bus_mode_addr,
                            l_bus_mode_reg),
             "Error from getScom (0x%08X)!", l_bus_mode_addr);
```
and `i_target` is type of `fapi2::TARGET_TYPE_PROC_CHIP`. [source](https://github.com/open-power/hostboot/blob/46a489ba7864b1d32c4b37a1223a39f33ad12f66/src/import/chips/p9/procedures/hwp/nest/p9_tod_utils.H#L313)
```
struct tod_topology_node
{
    fapi2::Target<fapi2::TARGET_TYPE_PROC_CHIP>* i_target;
    bool i_tod_master;
    bool i_drawer_master;
    p9_tod_setup_bus i_bus_rx; // Current node's bus from which step/sync is
    //   received ("Receive Port" in eiinfo)
    p9_tod_setup_bus i_bus_tx; // Upstream node's bus from which step/sync
    //   is transmitted ("Drive Port" in eiinfo)
    std::list<tod_topology_node*> i_children;
    p9_tod_setup_conf_regs o_todRegs;
    uint32_t o_int_path_delay;
};
```
Therefore it is not chiplet id, but **processor id**. For the time being we support only one processor, but I agree, this code could be future proofed. However looking in the documenatation or our own code I can't find any information about access to the other-processor's scom adress except for [this note](https://github.com/3mdeb/openpower-coreboot-docs/blob/main/devnotes/scom_access.md#assembly-instruction-ldcix) which i'm not sure if is correct.

If You know how to access the correct processor's scom, please let me know and I'll fix the discussed code.

Reply:
@krystian-hebel is this solution correct for You, or should we prepare the code somehow for a second processor in topology?

Reply:
Sorry for late response, I was trying to understand what happens in Hostboot for PROC target, but this is messed up... Leave it as it is right now, it will need a new function for second processor anyway. I think this is the first step we  are implementing that doesn't start with "for each functional PROC" loop at the top level.